nav2_util/include/nav2_util/simple_action_server.hpp

```diff
@@ -309,6 +309,24 @@ class SimpleActionServer
     return current_handle_->get_goal();
   }
 
+  /**
+   * @brief Terminate pending goals
+   */
+  void terminate_pending_goal()
+  {
+    std::lock_guard<std::recursive_mutex> lock(update_mutex_);
+
+    if (!pending_handle_ || !pending_handle_->is_active()) {
+      error_msg("Attempting to terminate pending goal when not available");
+      return;
+    }
+
+    terminate(pending_handle_);
```

Comment:
https://github.com/ros-planning/navigation2/blob/012fc19b059bf54a351809deb6709d523d3bfedd/nav2_util/include/nav2_util/simple_action_server.hpp#L304 should this also be reset after terminating?

Reply:
`terminate` resets the handle
https://github.com/ros-planning/navigation2/blob/012fc19b059bf54a351809deb6709d523d3bfedd/nav2_util/include/nav2_util/simple_action_server.hpp#L502